tests/components/charts.test.ts

```diff
@@ -566,31 +651,72 @@ describe("figures", () => {
         await simulateClick(".o-data-series input");
         setInputValueAndTrigger(".o-data-series input", "This is not valid", "input");
         await nextTick();
+        expect(fixture.querySelectorAll(".o-data-series .o-selection-ok").length).toBe(0);
+      }
+    );
+
+    test.each(["basicChart", "scorecard", "gauge"])(
+      "Clicking on reset button on dataset/keyValue/dataRange put back the last valid dataset/keyValue/dataRange",
+      async (chartType: string) => {
+        createTestChart(chartType);
+        await nextTick();
+
+        await simulateClick(".o-figure");
+        await simulateClick(".o-figure-menu-item");
+        await simulateClick(".o-menu div[data-name='edit']");
+        await simulateClick(".o-data-series input");
+        setInputValueAndTrigger(".o-data-series input", "A1", "input");
+        await nextTick();
         await simulateClick(".o-data-series .o-selection-ok");
-        expect(errorMessages()).toEqual(
-          expectedResults.map((result) => ChartTerms.Errors[result].toString())
+
+        await simulateClick(".o-data-series input");
+        setInputValueAndTrigger(".o-data-series input", "this is not valid", "input");
+        await nextTick();
+        await simulateClick(".o-data-series .o-selection-ko");
+
+        expect((fixture.querySelector(".o-data-series input") as HTMLInputElement).value).toBe(
+          "A1"
         );
       }
     );
 
-    test.each([
-      ["basicChart", [CommandResult.InvalidLabelRange]],
-      ["scorecard", [CommandResult.InvalidScorecardBaseline]],
-    ])(
-      "update chart with invalid labels/baseline",
-      async (chartType: string, expectedResults: CommandResult[]) => {
+    test.each(["basicChart", "scorecard"])(
+      "update chart with valid labels/baseline show confirm button/hide reset button",
+      async (chartType: string) => {
         createTestChart(chartType);
         await nextTick();
 
         await simulateClick(".o-figure");
         await simulateClick(".o-figure-menu-item");
         await simulateClick(".o-menu div[data-name='edit']");
         await simulateClick(".o-data-labels input");
-        setInputValueAndTrigger(".o-data-labels input", "this is not valid", "input");
+        setInputValueAndTrigger(".o-data-labels input", "A1", "input");
+        await nextTick();
+        expect(fixture.querySelectorAll(".o-data-labels .o-selection-ok").length).toBe(1);
+      }
+    );
+
+    test.each(["basicChart", "scorecard"])(
+      "update chart with invalid labels/baseline hide confirm button/show reset button",
```

Comment:
the confirm/show reset should also be tested in the selection_input component (actually, it makes more sense to test it without the chart context unless necessary)